crates/metadata-store/src/grpc/client.rs

```diff
@@ -286,6 +287,7 @@ fn map_status_to_write_error(status: Status) -> WriteError {
     match &status.code() {
         Code::Unavailable => WriteError::Network(status.into()),
         Code::FailedPrecondition => WriteError::FailedPrecondition(status.message().to_string()),
+        Code::Unimplemented => WriteError::RemoteError(RemoteError::Unimplemented),
         _ => WriteError::Internal(status.to_string()),
```

Comment:
What other errors could occur on the remote end that should be treated similarly? I am wondering whether introducing the special case of `Unimplemented` could be avoided by just having a generic `RemoteError`. How would `WriteError::Internal` then relate to this? Would there be the need to have `RemoteError` and `Internal` variants?

Reply:
While a generic RemoteError is very convenient, I thought it can make it harder to handle specific failure scenarios. But maybe better if we have a generic RemoteError that can still expose the failure status codes?.

For the Internal error. Is usually a catch all error type and I noticed that it's used for both remote and local errors that doesn't fit any of the other categories.